presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -266,7 +266,8 @@ public OrcBatchRecordReader createBatchRecordReader(
                 writeValidation,
                 initialBatchSize,
                 stripeMetadataSource,
-                cacheable);
+                cacheable,
+                orcReaderOptions);
```

Comment:
The members of OrcReaderOptions are not all meant for the StreamReaders. The `zstdJniDecompressionEnabled` is for the `decompressor` in `OrcReader`. In line 261-263, the 3 relevant members of the options are passed to the `OrcRecordReader` and `StreamReader`s already. Now you pass the whole option again. Can you take a look at how `legacyMapSubscript` is passed to the SelectiveXXXStreamReader's and use that way?

Reply:
@yingsu00 My intention was to avoid passing yet another boolean flag through multiple layers as I usually find it error prone: parameters of the same type may be passed in the wrong order and compiler won't complain. You're right that not all members OrcReaderOptions are required for StreamReaders (interface segregation principle violation) but I thought OrcReaderOptions is small enough to not be a concern for now. To make it a bit more attractive, I removed maxMergeDistance/tinyStripeThreshold/maxBlockSize ctor parameters in favor of OrcReaderOptions :)
Btw I don't have strong opinion so use boolean flag instead if it's your preference.